docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,408 @@ title: Data Grid - Editing
 
 # Data Grid - Editing
 
-<p class="description">The data grid has built-in edit capabilities that you can customize to your needs.</p>
+<p class="description">The data grid has built-in support for cell and row editing.</p>
 
-## Cell editing
+> ⚠️ This page refers to the new editing API, not enabled by default.
+> To use it, enable the following flag:
+>
+> ```tsx
+> <DataGrid experimentalFeatures={{ newEditingApi: true }} />
+> ```
+>
+> This additional step is required because the default editing API has a couple of issues which can only be fixed with breaking changes, only possible in v6.
+> To avoid having to wait for the next major release window, all breaking changes needed were included inside this flag.
+>
+> If you are looking for the documentation for the default editing API, visit [this page](/components/data-grid/editing-legacy/).
+> Note that it is encouraged to migrate to the new editing API since it will be enabled by default in v6.
+> Although it says "experimental", you can consider it as stable.
+
+## Making a column editable
 
-Cell editing allows editing the value of one cell at a time.
-Set the `editable` property in the `GridColDef` object to `true` to allow editing cells of this column.
+You can make a column editable by enabling the `editable` property in its [column definition](/api/data-grid/grid-col-def/):
 
 ```tsx
 <DataGrid columns={[{ field: 'name', editable: true }]} />
 ```
 
+This will enable the cell editing for the specified column.
+Note that there is also a row editing mode accessible by setting the `editMode` prop to `"row"`.
+For more information, see the dedicated [section](#row-editing).
+
+The following demo shows an example of how to make the Name column editable.
+Play with it by double-clicking or pressing <kbd class="key">Enter</kbd> in any cell from this column.
+
 {{"demo": "BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
 ### Start editing
 
-If a cell is editable and has focus, any of the following interactions will start the edit mode:
+These are the ways to start editing a cell (or row if `editMode="row"`):
```

Comment:
In general, I prefer to avoid the passive voice whenever possible. Try something like this to keep it conversational while bringing more clarity to who is performing the action:

> You can modify the filters through the grid interface in several different ways.

Applying that to the specific case of line 42 here, you could go with something like this:

> You can start editing a cell using any of the following interactions: